pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -1054,37 +1054,20 @@ private static int parseSql(char[] p_sql, int i, StringBuilder newsql, boolean s
             break;
           } else if (c == '{') { // start of an escape code?
             if (i + 1 < len) {
-              char next = p_sql[i + 1];
-              char nextnext = (i + 2 < len) ? p_sql[i + 2] : '\0';
-              if (next == 'd' || next == 'D') {
-                state = SqlParseState.ESC_TIMEDATE;
-                i++;
-                newsql.append("DATE ");
-                break;
-              } else if (next == 't' || next == 'T') {
-                state = SqlParseState.ESC_TIMEDATE;
-                if (nextnext == 's' || nextnext == 'S') {
-                  // timestamp constant
-                  i += 2;
-                  newsql.append("TIMESTAMP ");
-                } else {
-                  // time constant
-                  i++;
-                  newsql.append("TIME ");
+              SqlParseState[] availableStates = SqlParseState.values();
+              // skip first state, it's not a escape code state
+              for (int j = 1; j < availableStates.length; j++) {
+                SqlParseState availableState = availableStates[j];
+                int matched = SqlParseState.matchEscapeCode(p_sql, i + 1,
+                        availableState.escapeKeyword, availableState.allowedValues);
```

Comment:
Sorry for one-line comments, but wouldn't that be better to coded as "non-static" enum method?
Something like `int matchPoosition = availableState.getMatchPosition(p_sql, i+1)`.